packages/dev/core/src/Navigation/Plugins/recastJSv2/RecastNavigationJSPlugin.ts

```diff
@@ -0,0 +1,1770 @@
+/* eslint-disable no-console */
+
+import * as Recast2 from "recast-navigation";
+import { Matrix, Vector3 } from "../../../Maths/math.vector";
+import type { IAgentParameters, ICrowd, INavigationEnginePlugin, INavMeshParameters, IObstacle } from "../../INavigationEngine";
+import type { IVector3Like } from "../../../Maths/math.like";
+import { Mesh } from "../../../Meshes/mesh";
+import type { Nullable } from "../../../types";
+import { VertexBuffer } from "../../../Meshes/buffer";
+import { Logger } from "../../../Misc/logger";
+import type { Scene } from "../../../scene";
+import { VertexData } from "../../../Meshes/mesh.vertexData";
+import type { TransformNode } from "../../../Meshes/transformNode";
+import type { Observer } from "../../../Misc/observable";
+import { Observable } from "../../../Misc/observable";
+import { Epsilon } from "../../../Maths/math.constants";
+
+import type { SoloNavMeshGeneratorConfig, SoloNavMeshGeneratorIntermediates, TiledNavMeshGeneratorConfig, TiledNavMeshGeneratorIntermediates } from "recast-navigation/generators";
+import type { NavMesh, QueryFilter, TileCache } from "recast-navigation";
+import { generateSoloNavMesh, generateTileCache, generateTiledNavMesh } from "recast-navigation/generators";
+import { Crowd, Detour, exportNavMesh, getNavMeshPositionsAndIndices, getRandomSeed, importNavMesh, NavMeshQuery, setRandomSeed } from "recast-navigation";
```

Comment:
This is an interesting issue we have not solved yet. These types need to be declared and not imported, as it messes up the UMD declaration file. Is it optimal? nope. Is it safer for dev/core - yep :-)

Anything you need to import (and not import type) from the package needs to be injected. It can be an entire namespace, if needed.